src/System.Net.Primitives/src/System/Net/CredentialCache.cs

```diff
@@ -222,102 +221,169 @@ public NetworkCredential GetCredential(string host, int port, string authenticat
                 GlobalLog.Print("CredentialCache::GetCredential(host=\"" + host + ":" + port.ToString() + "\", authenticationType=\"" + authenticationType + "\")");
             }
 
-            NetworkCredential match = null;
-
-            // Enumerate through every credential in the cache
-            foreach (KeyValuePair<CredentialHostKey, NetworkCredential> pair in _cacheForHosts)
+            if (_cacheForHosts == null)
             {
-                CredentialHostKey key = pair.Key;
-
-                // Determine if this credential is applicable to the current Uri/AuthType
-                if (key.Match(host, port, authenticationType))
-                {
-                    match = pair.Value;
-                }
+                return null;
             }
 
+            var key = new CredentialHostKey(host, port, authenticationType);
+
+            NetworkCredential match;
```

Comment:
Nit: TryGetValue will return default(NetworkCredential) but the code would be much easy to read if match was initialized to null.